sycl/doc/syclcompat/README.md

```diff
@@ -947,6 +947,42 @@ public:
 };
 ```
 
+### ptr_to_int
+
+The following cuda backend specific function is introduced in order to
+translate from local memory pointers to `uint32_t` or `size_t` variables that
+contain a byte address to the local (local refers to`.shared` in nvptx) memory
+state space.
+
+``` c++
+namespace syclcompat {
+template <typename T>
+__syclcompat_inline__
+    std::enable_if_t<std::is_same_v<T, uint32_t> || std::is_same_v<T, size_t>,
+                     T>
+    ptr_to_int(void *ptr)
+} // syclcompat
+```
+
+These variables can be used in inline PTX instructions that take address
+operands. Such inline PTX instructions are commonly used in optimized
+libraries. A simplified example usage of the above functions is as follows:
+
+``` c++
+  half *data = syclcompat::local_mem<half[NUM_ELEMENTS]>();
+  // ...
+  // ...
+  T addr =
+              syclcompat::ptr_to_int<T>(reinterpret_cast<char *>(data) + (id % 8) * 16);
+
+uint32_t fragment;
+#if defined(__NVPTX__)
+  asm volatile("ldmatrix.sync.aligned.m8n8.x1.shared.b16 {%0}, [%1];\n"
+                : "=r"(fragment)
+                : "r"(addr));
+#endif
+```
```

Comment:
Could you fix the formatting of this code section? Thanks

Reply:
I did clang-format it already using dpc++ format.

Reply:
Possibly it's not running on code sections in markdown? I'd expect `uint32_t fragment` to align with `T addr` on the line above? And the line split on lines 975-976 looks pretty wacky? If I dump this code into a cpp file and autoformat this, I get:

```cpp
  half *data = syclcompat::local_mem<half[NUM_ELEMENTS]>();
  // ...
  // ...
  T addr =
      syclcompat::ptr_to_int<T>(reinterpret_cast<char *>(data) + (id % 8) * 16);
  uint32_t fragment;
#if defined(__NVPTX__)
  asm volatile("ldmatrix.sync.aligned.m8n8.x1.shared.b16 {%0}, [%1];\n"
               : "=r"(fragment)
               : "r"(addr));
#endif
```

Reply:
I can see if that passes clang-format (in the test where it is used). The existing version passes the clang-format on the clang-format CI.

Reply:
I don't think clang-format runs on the README tbh.

Reply:
Yeah, I use the same code in the test-e2e

Reply:
> Possibly it's not running on code sections in markdown? I'd expect `uint32_t fragment` to align with `T addr` on the line above? And the line split on lines 975-976 looks pretty wacky? If I dump this code into a cpp file and autoformat this, I get:
> 
> ```c++
>   half *data = syclcompat::local_mem<half[NUM_ELEMENTS]>();
>   // ...
>   // ...
>   T addr =
>       syclcompat::ptr_to_int<T>(reinterpret_cast<char *>(data) + (id % 8) * 16);
>   uint32_t fragment;
> #if defined(__NVPTX__)
>   asm volatile("ldmatrix.sync.aligned.m8n8.x1.shared.b16 {%0}, [%1];\n"
>                : "=r"(fragment)
>                : "r"(addr));
> #endif
> ```

I've updated the README with this suggestion now